xdsl/parser/attribute_parser.py

```diff
@@ -1262,6 +1263,11 @@ def _parse_optional_integer_or_float_type(self) -> Attribute | None:
             self._consume_token()
             return IndexType()
 
+        # Unit type
+        if name == "unit":
+            self._consume_token()
+            return UnitAttr()
+
```

Comment:
This is not the right place to add this though, this is neither an integer nor attribute type?